x-pack/plugins/observability/server/services/slo/get_slo_instances.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { ALL_VALUE, GetSLOInstancesResponse } from '@kbn/slo-schema';
+import { SLO_DESTINATION_INDEX_PATTERN } from '../../assets/constants';
+import { SLORepository } from './slo_repository';
+
+export class GetSLOInstances {
+  constructor(private repository: SLORepository, private esClient: ElasticsearchClient) {}
+
+  public async execute(sloId: string): Promise<GetSLOInstancesResponse> {
+    const slo = await this.repository.findById(sloId);
+
+    if (slo.groupBy === ALL_VALUE) {
+      return { groupBy: ALL_VALUE, instances: [] };
+    }
+
+    const result = await this.esClient.search({
+      index: SLO_DESTINATION_INDEX_PATTERN,
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            { range: { '@timestamp': { gte: 'now-7d' } } },
+            { term: { 'slo.id': slo.id } },
+            { term: { 'slo.revision': slo.revision } },
+          ],
+        },
+      },
+      aggs: {
+        instances: {
+          terms: {
+            size: 1000,
```

Comment:
I wonder if we shouldn't change this to `10_000` or a high number (within the max bucket size, 65K). The other option would be to use a `composite` aggregation and then paginate through all the responses, 5000 at a time.

Reply:
I think deferring this till we use the API makes sense.